sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs

```diff
@@ -22,8 +27,261 @@ namespace Azure.Messaging.EventHubs.Primitives
+                                 TokenCredential credential,
+                                 PartitionReceiverOptions options = default)
+        {
+            Argument.AssertNotNullOrEmpty(consumerGroup, nameof(consumerGroup));
+            Argument.AssertNotNullOrEmpty(partitionId, nameof(partitionId));
+            Argument.AssertNotNullOrEmpty(fullyQualifiedNamespace, nameof(fullyQualifiedNamespace));
+            Argument.AssertNotNullOrEmpty(eventHubName, nameof(eventHubName));
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            options = options?.Clone() ?? new PartitionReceiverOptions();
+
+            OwnsConnection = true;
+            Connection = new EventHubConnection(fullyQualifiedNamespace, eventHubName, credential, options.ConnectionOptions);
+            ConsumerGroup = consumerGroup;
+            PartitionId = partitionId;
+            InitialPosition = eventPosition;
+            RetryPolicy = options.RetryOptions.ToRetryPolicy();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionReceiver"/> class.
+        /// </summary>
+        ///
+        /// <param name="consumerGroup">The name of the consumer group this client is associated with.  Events are read in the context of this group.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition from which events will be received.</param>
+        /// <param name="eventPosition">The position within the partition where the client should begin reading events.</param>
+        /// <param name="connection">The <see cref="EventHubConnection" /> connection to use for communication with the Event Hubs service.</param>
+        /// <param name="options">A set of options to apply when configuring the client.</param>
+        ///
+        public PartitionReceiver(string consumerGroup,
+                                 string partitionId,
+                                 EventPosition eventPosition,
+                                 EventHubConnection connection,
+                                 PartitionReceiverOptions options = default)
+        {
+            Argument.AssertNotNullOrEmpty(consumerGroup, nameof(consumerGroup));
+            Argument.AssertNotNullOrEmpty(partitionId, nameof(partitionId));
+            Argument.AssertNotNull(connection, nameof(connection));
+
+            options = options?.Clone() ?? new PartitionReceiverOptions();
+
+            OwnsConnection = false;
+            Connection = connection;
+            ConsumerGroup = consumerGroup;
+            PartitionId = partitionId;
+            InitialPosition = eventPosition;
+            RetryPolicy = options.RetryOptions.ToRetryPolicy();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionReceiver"/> class.
+        /// </summary>
+        ///
+        protected PartitionReceiver()
+        {
+            OwnsConnection = false;
```

Comment:
On L87 we initialize this to `true`.  Then, in every constructor we either set it to `true` or `false`.  Do you think we could refactor this such that on L87 we default it to `false` and only set it to `true` in the two constructors where we create the actual connection?